Support automatic initial value for grid-auto-flow [css-grid-3][masonry]

fantasai opened the pull request:
#### ed04b373358b0dab92ff958721ae4ce3fd5d4307
<pre>
Support automatic initial value for grid-auto-flow [css-grid-3][masonry]
<a href="https://bugs.webkit.org/show_bug.cgi?id=302569">https://bugs.webkit.org/show_bug.cgi?id=302569</a>
<a href="https://rdar.apple.com/164791817">rdar://164791817</a>

Reviewed by Antti Koivisto and Brandon Stewart.

CSS Grid Level 3, which defines masonry layout, introduces a new initial
value for item-flow (which is essentially grid-auto-flow), which switches
the flow orientation automatically depending on the grid-template-rows &amp;
grid-template-columns properties:
- If only grid-template-columns is set, use waterfall layout.
- If only grid-template-rows is set, use brick layout.
- If both or neither are set, default to waterfall layout.

Implementing this provides an easy interface for authors, and also lets
us run most of the WPT tests for brick layout.

Test: fast/css-grid-layout/grid-auto-flow-get-set-grid-lanes.html
* LayoutTests/TestExpectations:
* LayoutTests/fast/css-grid-layout/grid-auto-flow-get-set-expected.txt:
* LayoutTests/fast/css-grid-layout/grid-auto-flow-get-set-grid-lanes-expected.txt: Added.
* LayoutTests/fast/css-grid-layout/grid-auto-flow-get-set-grid-lanes.html: Added.
* LayoutTests/fast/css-grid-layout/grid-auto-flow-get-set.html:
* LayoutTests/fast/css-grid-layout/grid-shorthand-get-set-expected.txt:
* LayoutTests/fast/css-grid-layout/grid-shorthand-get-set.html:
* LayoutTests/fast/css-grid-layout/resources/grid-definitions-parsing-utils.js:
(checkGridAutoFlowSetJSValue):
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-automatic-minimum-for-auto-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-flex-track-intrinsic-sizes-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-intrinsic-track-sizes-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-minimum-contribution-baseline-shim-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-minimum-contribution-with-percentages-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-minimum-size-grid-items-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-minimum-size-grid-items-003-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-007-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-008-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-009-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-010-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-011-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-017-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-018-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-022-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-023-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/parsing/grid-auto-flow-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/parsing/grid-template-columns-invalid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/parsing/grid-template-rows-invalid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssom-getPropertyValue-common-checks-expected.txt:
* LayoutTests/platform/ios/TestExpectations:

Pass more tests. Rebaseline a few. Add a new one for grid-lanes-specific
compute value conversions.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Grid.cpp:
(WebCore::CSSPropertyParserHelpers::consumeGridAutoFlow):
* Source/WebCore/layout/integration/grid/LayoutIntegrationGridCoverage.cpp:
(WebCore::LayoutIntegration::gridLayoutAvoidanceReason):
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::autoPlacementMajorAxisDirection const):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::initialGridAutoFlow):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setGridAutoFlowDirection):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):
* Source/WebCore/style/StyleExtractorCustom.h:
(WebCore::Style::PropertyExtractorAdaptor&lt;CSSPropertyGridAutoFlow&gt;::computedValue const):
(WebCore::Style::ExtractorCustom::extractGridAutoFlow):
(WebCore::Style::ExtractorCustom::extractGridAutoFlowSerialization):
* Source/WebCore/style/values/grid/StyleGridAutoFlow.cpp:
(WebCore::Style::CSSValueConversion&lt;GridAutoFlow&gt;::operator):
* Source/WebCore/style/values/grid/StyleGridAutoFlow.h:
(WebCore::Style::GridAutoFlow::GridAutoFlow):
(WebCore::Style::GridAutoFlow::setDirection):
(WebCore::Style::GridAutoFlow::switchOn const):
* Source/WebCore/style/values/grid/StyleGridTemplateList.h:
(WebCore::Style::GridTemplateList::isNone const):

Introduce a new &apos;normal&apos; initial value (not &apos;auto&apos; to avoid parsing
ambiguity in the &apos;grid&apos; shorthand) and make it compute to row or column
as appropriate.

Canonical link: <a href="https://commits.webkit.org/303634@main">https://commits.webkit.org/303634@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/668ef9bb83e8767d21c3efae0b6c80d4065d5730

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/133116 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5617 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44225 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140661 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/85155 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ca7776c1-2c2e-48cc-bd7c-97218f72a50f) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134986 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/6115 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5480 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101803 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/69186 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3796f4f1-1377-4257-a8a9-5579a89e996e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/136063 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/4316 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/119296 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82600 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/43004cc0-85ad-44f2-a375-469117ef6824) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/4200 "Passed tests") | [❌ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1783 "Found 1 new API test failure: TestWebKitAPI.ObscuredContentInsets.TopOverhangColorExtensionLayer (failure)") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/113275 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/37412 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143307 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5287 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37991 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110182 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5369 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4538 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110362 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27967 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4081 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115553 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58941 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5342 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33907 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5185 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68794 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5431 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5298 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

stwrt commented on Source/WebCore/style/StyleAdjuster.cpp:
```diff
@@ -566,6 +566,16 @@ void Adjuster::adjust(RenderStyle& style) const
+        if (style.gridAutoFlow().direction() == GridAutoFlow::Direction::Normal) {
+            // FIXME: Adjust this once CSSWG clarifies exactly how the initial value should compute on non-masonry.
+            // For now, this gives the most backwards-compatible behavior.
+            if ((style.display() == DisplayType::Masonry || style.display() == DisplayType::InlineMasonry)
```
This check is getting pretty hard to follow without careful reading.

Could we do something like:

```
bool isMasonry = style.display() == DisplayType::Masonry || style.display() == DisplayType::InlineMasonry;
bool emptyTemplateColumns = style.gridTemplateColumns().list.isEmpty();
bool emptyTemplateRows = style.gridTemplateRows().list.isEmpty();

if (isMasonry && emptyTemplateColumns && !emptyTemplateRows)
    style.setGridAutoFlowDirection(Style::GridAutoFlow::Direction::Column);
else
    style.setGridAutoFlowDirection(Style::GridAutoFlow::Direction::Row);
```

fantasai replied:
Added an isNone() helper function to StyleGridAutoFlow, rearranged the check a bit, and added comments roughly equivalent to the variable names you're proposing, lmk if that helps. The display check will also get simpler if we refactor DisplayType to separate inner and outer types.

Also, not sure if the compiler would optimize it away, but I wanted to short-circuit whatever calculations we can here, since it's irrelevant to most boxes...

stwrt replied:
It is better. I think we could use a function to determine if we have are a GridLane.

Something like the below in `LayoutBox.h`.

```C++
bool isGridLaneBox() {
   return style.display() == DisplayType::GridLanes || style.display() == DisplayType::InlineGridLanes;
}

```

fantasai replied:
I think this is something we should have more generally -- we need it for Grid and Flexbox, too, and we need it in many places across WebKit -- so my thoughts are to handle it in a follow-up patch to clean up `DisplayType` checks more generally.

stwrt replied:
Sounds good

stwrt reviewed (commented):
It looks like all the Linux CIs are failing.

```
FAILED: Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-198.cpp.o 
/usr/bin/ccache /usr/bin/c++ -DBITMAP_TEXTURE_POOL_MAX_SIZE_IN_MB=80 -DBUILDING_WEBKIT=1 -DBUILDING_WITH_CMAKE=1 -DBUILDING_WPE__=1 -DBUILDING_WebCore -DBWRAP_EXECUTABLE=\"/usr/bin/bwrap\" -DDBUS_PROXY_EXECUTABLE=\"/usr/bin/xdg-dbus-proxy\" -DGETTEXT_PACKAGE=\"WPE\" -DHAVE_CONFIG_H=1 -DJSC_GLIB_API_ENABLED -DPAS_BMALLOC=1 -DSK_ASSUME_GL=0 -DSK_ASSUME_GL_ES=1 -DSK_ASSUME_WEBGL=0 -DSK_DISABLE_LEGACY_GL_MAKE_NATIVE_INTERFACE -DSK_DISABLE_LEGACY_IMAGE_READBUFFER -DSK_DISABLE_LEGACY_INIT_DECODERS -DSK_DISABLE_LEGACY_PNG_WRITEBUFFER -DSK_DISABLE_TRACING -DSK_DISABLE_WEBGL_INTERFACE -DSK_ENABLE_PRECOMPILE -DSK_R32_SHIFT=16 -DSK_RELEASE -DSTATICALLY_LINKED_WITH_JavaScriptCore -DSTATICALLY_LINKED_WITH_PAL -DSTATICALLY_LINKED_WITH_WTF -DSTATICALLY_LINKED_WITH_bmalloc -DTHUNDER_HAS_OCDM_DECRYPT_BUFFER=0 -DUSE_SYSTEM_EGL -D_GLIBCXX_ASSERTIONS=1 -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/WebCore/PrivateHeaders -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/WebCore/DerivedSources -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/ShapeDetection -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/ShapeDetection/Interfaces -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/WebGPU -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/WebGPU/InternalAPI -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/WebGPU/Implementation -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/airplay -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/applepay -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/applepay/paymentrequest -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/applicationmanifest -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/async-clipboard -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/audiosession -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/badge -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/beacon -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/cache -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/compression -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/contact-picker -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/cookie-consent -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/cookie-store -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/credentialmanagement -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/encryptedmedia -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/encryptedmedia/legacy -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/entriesapi -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/fetch -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/filesystem -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/geolocation -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/highlight -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/identity -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/identity/dummy -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/identity/protocols -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/identity/protocols/ISO18013 -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/identity/protocols/openid -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/indexeddb -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/indexeddb/client -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/indexeddb/server -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/indexeddb/shared -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/mediacapabilities -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/mediacontrols -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/mediarecorder -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/mediasession -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/mediasource -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/mediastream -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/model-element -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/model-element/dummy -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/navigatorcontentutils -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/notifications -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/paymentrequest -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/permissions -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/pictureinpicture -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/plugins -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/push-api -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/remoteplayback -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/reporting -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/screen-wake-lock -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/speech -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/storage -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/streams -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/url-pattern -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/web-locks -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/webaudio -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/webauthn -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/webauthn/cbor -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/webauthn/fido -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/webcodecs -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/webdatabase -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/webdriver -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/websockets -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/webtransport -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/webxr -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/accessibility -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/accessibility/isolatedtree -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/animation -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/automation -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/bindings -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/bindings/js -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/bridge -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/bridge/c -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/bridge/jsc -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/contentextensions -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/crypto -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/crypto/algorithms -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/crypto/keys -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/crypto/parameters -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/calc -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/parser -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/query -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/typedom -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/typedom/color -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/typedom/numeric -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/typedom/transform -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/borders -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/color -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/color-adjust -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/easing -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/filter-effects -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/grid -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/images -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/motion -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/primitives -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/shapes -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/css/values/text-decoration -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/cssjit -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/dom -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/dom/messageports -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/domjit -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/editing -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/fileapi -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/history -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/html -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/html/canvas -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/html/closewatcher -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/html/forms -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/html/parser -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/html/shadow -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/html/track -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/inspector -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/inspector/agents -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/inspector/agents/page -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/inspector/agents/worker -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts/block -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts/block/tablewrapper -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts/flex -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts/grid -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/floats -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts/inline -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts/inline/display -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts/inline/invalidation -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts/inline/ruby -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts/inline/text -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/integration -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/integration/flex -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/integration/grid -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/integration/inline -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/layouttree -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/layout/formattingContexts/table -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/loader -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/loader/archive -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/loader/archive/mhtml -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/loader/cache -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/loader/icon -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/mathml -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/page -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/page/csp -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/page/scrolling -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/page/text-extraction -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/animation -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/animation/values -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/audio -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/encryptedmedia -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/gamepad -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/cpu/arm -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/cpu/arm/filters -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/displaylists -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/filters -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/filters/software -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/controls -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/iso -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/opentype -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/transforms -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/mediacapabilities -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/mediarecorder -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/mediastream -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/mock -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/mock/mediasource -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/network -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/sql -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/text -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/xr -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/plugins -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/rendering -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/rendering/line -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/rendering/mathml -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/rendering/shapes -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/rendering/style -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/rendering/svg -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/rendering/svg/legacy -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/rendering/updating -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/replay -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/storage -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/calc -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/align -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/anchor-position -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/animations -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/backgrounds -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/borders -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/box -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/break -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/color-adjust -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/color -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/contain -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/content -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/counter-styles -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/css2 -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/display -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/easing -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/fill-stroke -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/filter-effects -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/flexbox -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/fonts -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/grid -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/images -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/inline -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/line-grid -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/lists -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/masking -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/motion -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/multicol -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/non-standard -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/overflow -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/page -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/pointerevents -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/position -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/primitives -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/rhythm -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/scroll-animations -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/scroll-snap -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/scrollbars -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/shapes -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/size-adjust -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/sizing -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/svg -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/text -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/text-decoration -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/transforms -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/transforms/functions -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/transitions -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/ui -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/view-transitions -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/style/values/will-change -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/svg -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/svg/animation -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/svg/graphics -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/svg/graphics/filters -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/svg/properties -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/websockets -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/workers -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/workers/service -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/workers/service/background-fetch -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/workers/service/context -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/workers/service/server -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/workers/shared -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/workers/shared/context -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/worklets -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/xml -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/xml/parser -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/gamepad -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/encryptedmedia/clearkey -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/angle -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/lcms -I/home/buildbot-worker/WPE-Build-EWS/build/Source/ThirdParty/xdgmime/src -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/adwaita -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/adwaita -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/harfbuzz -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/filters/skia -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/skia -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/Modules/mediastream/gstreamer -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/gstreamer -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/gstreamer/mse -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/gstreamer/eme -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/gstreamer/telemetry -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/gstreamer -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/mediarecorder/gstreamer -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/audio/gstreamer -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/image-decoders -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/image-decoders/avif -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/image-decoders/bmp -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/image-decoders/gif -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/image-decoders/ico -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/image-decoders/jpeg -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/image-decoders/jpegxl -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/image-decoders/png -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/image-decoders/webp -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/network/soup -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/texmap -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/page/scrolling/coordinated -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/texmap/coordinated -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/accessibility/atspi -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/crypto/openssl -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/audio/glib -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/glib -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/android -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/egl -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/epoxy -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/gbm -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/opengl -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/libwpe -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/wayland -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/mediastream/gstreamer -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/network/glib -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/text/icu -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/video-codecs -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/wpe -I/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/gamepad/libwpe -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/ANGLE/Headers -I/home/buildbot-worker/WPE-Build-EWS/build/Source/ThirdParty/skia -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/Skia/Headers/top -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/Skia/Headers/mod -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/JavaScriptCore/Headers -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/JavaScriptCore/PrivateHeaders -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/JavaScriptCoreGLib/Headers -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/JavaScriptCoreGLib/DerivedSources -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/WTF/Headers -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/bmalloc/Headers -I/home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/PAL/Headers -isystem /jhbuild/install/include/gstreamer-1.0 -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/include/libdrm -isystem /jhbuild/install/include/orc-0.4 -isystem /jhbuild/install/lib/gstreamer-1.0/include -isystem /usr/include/gudev-1.0 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /jhbuild/install/include/WPEFramework/ocdm -isystem /usr/include/flite -isystem /usr/include/libxml2 -isystem /usr/include/harfbuzz -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/avif -isystem /jhbuild/install/include/libsoup-3.0 -isystem /jhbuild/install/include/wpe-1.0 -isystem /jhbuild/install/include/sysprof-6 -DASSERT_ENABLED=1 -fdiagnostics-color=always -Wextra -Wall -pipe -fmax-errors=20 -Wno-odr -Wno-stringop-overread -Wno-stringop-overflow -Wno-nonnull -Wno-array-bounds -Wno-expansion-to-defined -Wno-noexcept-type -Wno-subobject-linkage -Wno-psabi -Wno-misleading-indentation -Wno-maybe-uninitialized -Wundef -Wpointer-arith -Wmissing-format-attribute -Wformat-security -Wcast-align -Wno-tautological-compare -fasynchronous-unwind-tables -fdebug-types-section -ffile-prefix-map=/home/buildbot-worker/WPE-Build-EWS/build=. -fno-omit-frame-pointer -fno-strict-aliasing -fno-exceptions -fno-rtti -fcoroutines -ffunction-sections -fdata-sections -O3 -DNDEBUG -std=c++23 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Werror -DHAVE_HB_FEATURES_H -I/usr/include/gio-unix-2.0 -I/usr/include -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -pthread -I/usr/include/libmount -I/usr/include/blkid -DHWY_SHARED_DEFINE -DAVIF_DLL -DWPE_ENABLE_XKB=1 -MD -MT Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-198.cpp.o -MF Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-198.cpp.o.d -o Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-198.cpp.o -c /home/buildbot-worker/WPE-Build-EWS/build/WebKitBuild/WPE/Release/WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-198.cpp
In file included from WebCore/DerivedSources/StyleExtractorGenerated.cpp:11,
                 from WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-198.cpp:4:
../../../Source/WebCore/style/StyleExtractorCustom.h: In member function ‘decltype(auto) WebCore::Style::PropertyExtractorAdaptor<WebCore::CSSPropertyGridAutoFlow>::computedValue(WebCore::Style::ExtractorState&, F&&) const [with F = WebCore::Style::extractCSSValue<WebCore::CSSPropertyGridAutoFlow>(ExtractorState&)::<lambda(auto:533&&)>]’:
../../../Source/WebCore/style/StyleExtractorCustom.h:862:5: error: control reaches end of non-void function [-Werror=return-type]
  862 |     }
      |     ^
cc1plus: all warnings being treated as errors
```

stwrt reviewed (approved):
LGTM if tests pass

fantasai committed:
Support automatic initial value for grid-auto-flow [css-grid-3][masonry]
https://bugs.webkit.org/show_bug.cgi?id=302569
rdar://164791817

Reviewed by Antti Koivisto and Brandon Stewart.

CSS Grid Level 3, which defines masonry layout, introduces a new initial
value for item-flow (which is essentially grid-auto-flow), which switches
the flow orientation automatically depending on the grid-template-rows &
grid-template-columns properties:
- If only grid-template-columns is set, use waterfall layout.
- If only grid-template-rows is set, use brick layout.
- If both or neither are set, default to waterfall layout.

Implementing this provides an easy interface for authors, and also lets
us run most of the WPT tests for brick layout.

Test: fast/css-grid-layout/grid-auto-flow-get-set-grid-lanes.html
* LayoutTests/TestExpectations:
* LayoutTests/fast/css-grid-layout/grid-auto-flow-get-set-expected.txt:
* LayoutTests/fast/css-grid-layout/grid-auto-flow-get-set-grid-lanes-expected.txt: Added.
* LayoutTests/fast/css-grid-layout/grid-auto-flow-get-set-grid-lanes.html: Added.
* LayoutTests/fast/css-grid-layout/grid-auto-flow-get-set.html:
* LayoutTests/fast/css-grid-layout/grid-shorthand-get-set-expected.txt:
* LayoutTests/fast/css-grid-layout/grid-shorthand-get-set.html:
* LayoutTests/fast/css-grid-layout/resources/grid-definitions-parsing-utils.js:
(checkGridAutoFlowSetJSValue):
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-automatic-minimum-for-auto-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-flex-track-intrinsic-sizes-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-intrinsic-track-sizes-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-minimum-contribution-baseline-shim-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-minimum-contribution-with-percentages-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-minimum-size-grid-items-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/items/row-minimum-size-grid-items-003-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-007-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-008-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-009-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-010-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-011-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-017-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-018-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-022-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-lanes/tentative/track-sizing/auto-repeat/row-auto-repeat-023-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/parsing/grid-auto-flow-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/parsing/grid-template-columns-invalid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/parsing/grid-template-rows-invalid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssom-getPropertyValue-common-checks-expected.txt:
* LayoutTests/platform/ios/TestExpectations:

Pass more tests. Rebaseline a few. Add a new one for grid-lanes-specific
compute value conversions.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Grid.cpp:
(WebCore::CSSPropertyParserHelpers::consumeGridAutoFlow):
* Source/WebCore/layout/integration/grid/LayoutIntegrationGridCoverage.cpp:
(WebCore::LayoutIntegration::gridLayoutAvoidanceReason):
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::autoPlacementMajorAxisDirection const):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::initialGridAutoFlow):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setGridAutoFlowDirection):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):
* Source/WebCore/style/StyleExtractorCustom.h:
(WebCore::Style::PropertyExtractorAdaptor<CSSPropertyGridAutoFlow>::computedValue const):
(WebCore::Style::ExtractorCustom::extractGridAutoFlow):
(WebCore::Style::ExtractorCustom::extractGridAutoFlowSerialization):
* Source/WebCore/style/values/grid/StyleGridAutoFlow.cpp:
(WebCore::Style::CSSValueConversion<GridAutoFlow>::operator):
* Source/WebCore/style/values/grid/StyleGridAutoFlow.h:
(WebCore::Style::GridAutoFlow::GridAutoFlow):
(WebCore::Style::GridAutoFlow::setDirection):
(WebCore::Style::GridAutoFlow::switchOn const):
* Source/WebCore/style/values/grid/StyleGridTemplateList.h:
(WebCore::Style::GridTemplateList::isNone const):

Introduce a new 'normal' initial value (not 'auto' to avoid parsing
ambiguity in the 'grid' shorthand) and make it compute to row or column
as appropriate.

Canonical link: https://commits.webkit.org/303634@main
